Gems/RemoteTools/Code/Source/Utilities/RemoteToolsJoinThread.cpp

```diff
@@ -0,0 +1,97 @@
+
+namespace RemoteTools
+{
+    RemoteToolsJoinThread::RemoteToolsJoinThread(int updateRate, RemoteToolsSystemComponent* component)
+        : TimedThread("RemoteTools::RemoteToolsJoinThread", AZ::TimeMs(updateRate))
+    {
+        m_remoteToolsComponent = component;
+    }
+
+    RemoteToolsJoinThread::~RemoteToolsJoinThread()
+    {
+        Stop();
+        Join();
+    }
+
+    void RemoteToolsJoinThread::OnUpdate([[maybe_unused]] AZ::TimeMs updateRateMs)
+    {
+        bool isRequestingConnection = false;
+        auto& networkInterfaces = AZ::Interface<AzNetworking::INetworking>::Get()->GetNetworkInterfaces();
+        for (auto registryIt = m_remoteToolsComponent->m_entryRegistry.begin();
+            registryIt != m_remoteToolsComponent->m_entryRegistry.end();
+             ++registryIt)
+        {
+            AZ::Name serviceName = registryIt->second.m_name;
+            if (!registryIt->second.m_isHost && networkInterfaces.contains(serviceName))
+            {
+                auto networkInterface = networkInterfaces.find(serviceName);
+                if (networkInterface->second->GetConnectionSet().GetActiveConnectionCount() == 0)
+                {
+                    AzNetworking::ConnectionId connId = networkInterface->second->Connect(registryIt->second.m_ip);
+                    if (connId != AzNetworking::InvalidConnectionId)
+                    {
+                        RemoteToolsPackets::RemoteToolsConnect initPacket;
+                        initPacket.SetPersistentId(registryIt->first);
+                        initPacket.SetDisplayName(serviceName.GetCStr());
+                        networkInterface->second->SendReliablePacket(connId, initPacket);
+                    }
+                    else
+                    {
+                        isRequestingConnection = true;
+                    }
+                }
+            }
+        }
+
+        if (!isRequestingConnection)
+        {
+            Stop();
+            Join();
+        }
+    }
+
+    void RemoteToolsJoinThread::UpdateStatus()
+    {
+        if (!IsRunning())
+        {
+            bool isRequestingConnection = false;
+            auto& networkInterfaces = AZ::Interface<AzNetworking::INetworking>::Get()->GetNetworkInterfaces();
+            for (auto registryIt = m_remoteToolsComponent->m_entryRegistry.begin();
+                 registryIt != m_remoteToolsComponent->m_entryRegistry.end();
+                 ++registryIt)
+            {
+                AZ::Name serviceName = registryIt->second.m_name;
+                if (networkInterfaces.contains(serviceName))
+                {
+                    auto networkInterface = networkInterfaces.find(serviceName);
+                    if (networkInterface->second->GetConnectionSet().GetActiveConnectionCount() == 0)
+                    {
+                        isRequestingConnection = true;
+                        break;
+                    }
+                }
+            }
```

Comment:
Why not just use a range based for loop for this?

```suggestion
            for (const RemoteToolsRegistryEntry& toolsRegistryEntry : AZStd::ranges::views::values(m_remoteToolsComponent->m_entryRegistry))
            {
                AZ::Name serviceName = toolsRegistryEntry.m_name;
                if (networkInterfaces.contains(serviceName))
                {
                    auto networkInterface = networkInterfaces.find(serviceName);
                    if (networkInterface->second->GetConnectionSet().GetActiveConnectionCount() == 0)
                    {
                        isRequestingConnection = true;
                        break;
                    }
                }
            }
```